shotover/src/transforms/kafka/sink_cluster/shotover_node.rs

```diff
@@ -33,9 +40,85 @@ pub(crate) struct ShotoverNode {
     state: Arc<AtomicShotoverNodeState>,
 }
 
+impl ShotoverNode {
+    #![allow(unused)]
+    pub(crate) fn is_up(&self) -> bool {
+        self.state.load(Ordering::Relaxed) == ShotoverNodeState::Up
+    }
+
+    pub(crate) fn set_state(&self, state: ShotoverNodeState) {
+        self.state.store(state, Ordering::Relaxed)
+    }
+}
+
 #[atomic_enum]
 #[derive(PartialEq)]
 pub(crate) enum ShotoverNodeState {
     Up,
     Down,
 }
+
+pub(crate) fn start_shotover_peers_check(
+    shotover_peers: Vec<ShotoverNode>,
+    check_shotover_peers_delay_ms: u64,
+    connect_timeout_ms: u64,
+) {
+    if !shotover_peers.is_empty() {
+        tokio::spawn(async move {
+            // Wait for all shotover nodes to start
+            sleep(Duration::from_secs(10)).await;
+            loop {
+                match check_shotover_peers(
+                    &shotover_peers,
+                    check_shotover_peers_delay_ms,
+                    connect_timeout_ms,
+                )
+                .await
+                {
+                    Ok(_) => {}
+                    Err(err) => {
+                        tracing::error!(
+                            "Restarting the shotover peers check due to error: {err:?}"
+                        );
+                    }
+                };
+            }
+        });
+    }
+}
+
+async fn check_shotover_peers(
+    shotover_peers: &[ShotoverNode],
+    check_shotover_peers_delay_ms: u64,
+    connect_timeout_ms: u64,
```

Comment:
I agree that check_shotover_peers_delay_ms should stay as is so we can easily perform the +- random jitter.
However, `connect_timeout_ms: u64` should be converted to a `connect_timeout: Duration` earlier, in `KafkaSinkClusterBuilder::new` would be reasonable.
When there is a stronger typed representation available its better to make use of that as much as we can.